src/core/bpf/allow_bind/allow-bind.c

```diff
@@ -0,0 +1,41 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <bpf/bpf_endian.h>
+#include <linux/bpf.h>
+#include <netinet/in.h>
+
+#include "maps.h"
+
+static inline int allow_bind_impl(
+                struct bpf_sock_addr* ctx,
+                struct allow_ports_t* allowed_ports) {
+        __u16 port_key;
+        __u8* port;
+
+        volatile __u32 user_port = ctx->user_port;
+        port_key = (__u16)bpf_ntohs(user_port);
+        port = bpf_map_lookup_elem(allowed_ports, &port_key);
+        if (!port)
+                return 0;
+
+        return 1;
+}
+
+SEC("cgroup/bind6")
+int allow_bind_v6(struct bpf_sock_addr* ctx) {
+        if (ctx->user_family != AF_INET6 || ctx->family != AF_INET6)
+                return 1;
+
+        return allow_bind_impl(ctx, &allow_ports);
+}
+
+SEC("cgroup/bind4")
+int allow_bind_v4(struct bpf_sock_addr* ctx) {
+        if (ctx->user_family != AF_INET || ctx->family != AF_INET)
+                return 1;
+
+        return allow_bind_impl(ctx, &allow_ports);
+}
+
+int _version SEC("version") = 1;
```

Comment:
this is not necessary for a long time, and even on very old kernels that required version, it was only used for kprobes, I believe. So feel free to drop this.